libethashseal/Ethash.cpp

```diff
@@ -152,8 +152,13 @@ void Ethash::verifyTransaction(ImportRequirements::value _ir, TransactionBase co
 	{
 		if (_bi.number() >= chainParams().u256Param("EIP158ForkBlock"))
 		{
-			int chainID(chainParams().u256Param("chainID"));
-			_t.checkChainId(chainID);
+			if (_bi.number() >= chainParams().u256Param("metropolisForkBlock") && _t.hasZeroSignature())
+				_t.checkChainId(0);
```

Comment:
I think that's fine to keep the previous code
`
int chainID(chainParams().u256Param("chainID"));
_t.checkChainId(chainID);
`

in the constructor of `TransactionBase`, m_chain is set to sign.v in case of zero signature